packages/multi-test/src/app.rs

```diff
@@ -903,4 +911,172 @@ mod test {
             .unwrap();
         assert_eq!(state.beneficiary, random);
     }
+
+    mod replay_data_overwrite {
+        use cosmwasm_std::to_binary;
+
+        use crate::test_helpers::EmptyMsg;
+
+        use super::*;
+
+        fn make_echo_submsg(
+            contract: Addr,
+            data: impl Into<Option<&'static str>>,
+            sub_msg: Vec<SubMsg<Binary>>,
+        ) -> SubMsg<Binary> {
+            let data = data.into().map(|s| s.to_owned());
+            SubMsg::new(CosmosMsg::Wasm(WasmMsg::Execute {
+                contract_addr: contract.into(),
+                msg: to_binary(&echo::Message { data, sub_msg }).unwrap(),
+                funds: vec![],
+            }))
+        }
+
+        #[test]
+        fn no_submsg() {
+            let mut app = binary_app();
+
+            let owner = Addr::unchecked("owner");
+
+            let contract_id = app.store_code(echo::contract());
+            let contract = app
+                .instantiate_contract(contract_id, owner.clone(), &EmptyMsg {}, &[], "Echo", None)
+                .unwrap();
+
+            let response = app
+                .execute_contract(
+                    owner,
+                    contract,
+                    &echo::Message {
+                        data: Some("Data".to_owned()),
+                        sub_msg: vec![],
+                    },
+                    &[],
+                )
+                .unwrap();
+
+            assert_eq!(response.data, Some("Data".to_owned().into_bytes().into()));
```

Comment:
That is some chain... Doesn't it work without the `to_owned`?

Or `"Data".to_bytes().into()`?